contracts/ROSCA.sol

```diff
@@ -0,0 +1,224 @@
+pragma solidity ^0.4.4;
+
+/**
+*  Things still missing/needs fixing:
+*   in withdraw() , add fee related logic
+*   checkingContribution, check whether or not a member as contributed as least contributionSize at the end of Round
+*/
+contract ROSCA {
+  address constant WETRUST = 0x0;
+  uint constant MIN_CONTRIBUTION_SIZE = 1000000000000;  // 1e12
+  uint constant MAX_FEE_IN_THOUSANDTH = 200;
+  address constant WETRUST_FEE_ADDRESS = 0x0;
+  uint constant MINIMUM_TIME_BEFORE_ROSCA_START = 1 days;
+  uint constant MINIMUM_PARTICIPANTS = 2;
+  uint16 constant EPOCH_TO_RUN = 1;
+  uint constant MIN_ROUND_PERIOD_IN_DAYS = 1;
+  uint constant MAX_ROUND_PERIOD_IN_DAYS = 30;
+  uint constant MIN_DISTRIBUTION_RATIO = 65;
+
+  event LogParticipantApplied(address user);
+  event LogParticipantApprove(address user);
+  event LogContributionMade(address user, uint amount);
+  event LogNewLowestBid(uint bid,address winnerAddress);
+  event LogRoundFundsReleased(address winnerAddress, uint amountInWei);
+  event LogFundsWithdrawal(address user, uint amount,address destination);
+  event LogStartOfPeriod(uint currentRound);
+
+  // state variables
+  //uint16 numRoundsInEpoch;     // number of rounds in 1 epoch, user doesnt have control over this, it'll increment everytime a new participant is registered
```

Comment:
nit: removing dead code can help with understandability and thus security of the code. We bring it back (from old snapshots?) when needed.